### PR TITLE
230 us additional views navigation

### PR DESCRIPTION
**Pull request by witnick**
I abstracted away the schema selector in the top bar and the one in the setting into a single component( see SchemaSelector.tsx) since they have the same behaviour.

I also changed the layout of the accordion that display the views at Eddie's request. It is no longer showing the views based on the table but instead it is based on the schema. 

Small demo:
additionalview

I created an AdditionalViewNavBar component. When a table has additional views available it will show with the different options, otherwise it is hidden. 
For the time being it only shows the available display type and the links don't lead to a view. Future stories will be the implementation of the views

image

### CODE REVIEW

**Review by AfandiM — DISMISSED** [review]
It looks good to me. Given the size of the PR, I would recommend an additional review from someone else, but 👍

File: UInnovateApp/src/pages/Settings.tsx [hunk 1]
```diff
@@ -51,9 +53,9 @@ export function Settings() {
 		dispatch(updateSelectedSchema(selectedSchema));
 	}, [selectedSchema])
 
-	const handleSchemaChange = (event: SelectChangeEvent) => {
-		setSelectedSchema(event.target.value);
-	};
+	// const handleSchemaChange = (event: SelectChangeEvent) => {
```
Comment (lyubageorgieva) [inline UInnovateApp/src/pages/Settings.tsx:56]: Do you need all of this commented out?

File: UInnovateApp/src/pages/Settings.tsx [hunk 2]
```diff
@@ -77,7 +79,7 @@ export function Settings() {
 					<div className='save-config-container'>
 						<Box display="flex" gap={"2rem"} alignItems={"center"}>
 							<h1 className='title'>Settings</h1>
-							<FormControl fullWidth disabled={schemas.length === 0}>
+							{/* <FormControl fullWidth disabled={schemas.length === 0}>
```
Comment (lyubageorgieva) [inline UInnovateApp/src/pages/Settings.tsx:82]: This also?

File: UInnovateApp/src/virtualmodel/AdditionalViewsDataAccessor.ts [hunk 3]
```diff
@@ -72,7 +82,7 @@ export const insertNewView = async (p_schemaName: string, p_tableName: string, p
 
 			await cstm_view_data_accessor.executeFunction();
 		}
-	
+
 		// alert("View saved successfully.");
```
Comment (lyubageorgieva) [inline UInnovateApp/src/virtualmodel/AdditionalViewsDataAccessor.ts:86]: The comment here also?

File: UInnovateApp/src/components/AdditionalViewNavBar.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+const AdditionalViewNavBar = ({selectedSchema, selectedTable}: AdditionalViewNavBarProp) => {
+    const [viewList, setViewList] = useState<AdditionalViews[]>([]);
+    const [customViews, setCustomViews]= useState([]);
+
+    useEffect(()=>{
+        const ctrl = new AbortController();
+        const signal = ctrl.signal;
+        // get data from db
+        getViewsBySchema(setViewList, selectedSchema, signal);
+        // getCustomViews(setCustomViews, signal)
```
Comment (lyubageorgieva) [inline UInnovateApp/src/components/AdditionalViewNavBar.tsx:21]: Do we need this?

**Review by lyubageorgieva — CHANGES_REQUESTED** [review]
Is it possible to fix the buttons, they are all lower caps while if you look at our other tabs it's all caped, and also with their colours in the modal. They will be a general refactoring later on but if it's possible to fix the letter case at least and the colours from now it would help for later on 
Screenshot 2024-02-24 at 9 49 53 AM

**Review by lyubageorgieva — CHANGES_REQUESTED** [review]
Here I see View three times, and I find it not easy on the eyes.

Screenshot 2024-02-24 at 9 55 56 AM


Maybe instead, just add ADD VIEW without the title and the "no view"
Screenshot 2024-02-24 at 9 59 18 AM

**Review by lyubageorgieva — CHANGES_REQUESTED** [review]
Here also the button "delete"
Screenshot 2024-02-24 at 10 01 14 AM

**Comment by witnick** [comment]
I changed the style of the buttons and removed the View title

additionalview_icons

File: UInnovateApp/src/components/AdditionalViewNavBar.tsx (hunk 4 above)
Comment (witnick) [reply UInnovateApp/src/components/AdditionalViewNavBar.tsx:21]: yes that is actually the requirement for #230.

**Pull request merged**